### ISSUE TITLE
vim text area `CTRL+I` not working on some text areas

### ISSUE THREAD

**Issue by joech4n** [brookhong/Surfingkeys#140]
It looks like it doesn't work in a gist.github.com main text area. It seems to work in other smaller (one line) text areas, however (such as this Github issue).

**Comment by brookhong** [comment]
I looked into this, gist.github.com uses codemirror for the main text area. We need a proper way to set value of the editor, but extension has its own context different than the content window for security.

**Comment by tlegutko** [comment]
It also doens't seem to work on Google's Inbox, when composing email. It work's in Inboxes search bar (one liner) though.

**Comment by colossatr0n** [comment]
@brookhong Setting the text value of the CodeMIrror editor can be done like this:
```js
var codeMirrorEditor = document.querySelector( '.CodeMirror' ).CodeMirror;
codeMirrorEditor.setValue("text from the vim editor window")
```

I'd be interested in looking into this if you can point me in the right direction.

I'm imagining that Surfingkeys could do a CSS class check on the text area element when `:w` is issued to see if it's a CodeMirror element, and if so, insert the text using the code shown above instead of doing it the typical way.

**Comment by brookhong** [comment]
You can start it from https://github.com/brookhong/Surfingkeys/blob/3999da774f409f3756b5f6109be8092eda8dd9b8/src/content_scripts/front.js#L228-L248

But the problem is that the code snippet you provided above only works in the main world(aka the scope of the page itself), it won't work in extension's space.

**Comment by colossatr0n** [comment]
Thanks for the quick response. I've been trying various things to get this to work but haven't had any success. It looks like the only way to edit CodeMirror text is through their API, which isn't possible as you've pointed out. Editing the DOM directly using the same element structure that their API generates also doesn't work.

**Comment by colossatr0n** [comment]
I actually found a solution and implemented it in PR #1921.
